testcases/OpTestKexec.py

```diff
@@ -0,0 +1,161 @@
+from common.OpTestError import OpTestError
+
+log = OpTestLogger.optest_logger_glob.get_logger(__name__)
+
+class OpTestKexec(unittest.TestCase):
+
+    def setUp(self):
+        conf = OpTestConfiguration.conf
+        self.cv_SYSTEM = conf.system()
+        self.c = self.cv_SYSTEM.console
+        self.cv_HOST = conf.host()
+        self.num_of_iterations = conf.args.num_of_iterations
+        self.kernel_image = conf.args.kernel_image
+        self.initrd_image = conf.args.initrd_image
+        self.linux_src_dir = conf.args.linux_src_dir
+        if not (self.kernel_image and self.initrd_image):
+            if self.linux_src_dir:
+                kernel_release = self.cv_HOST.host_run_command("cd {} && make kernelrelease".format(self.linux_src_dir))[0]
+            else:
+                kernel_release = self.cv_HOST.host_run_command("uname -r")[0]
+            self.kernel_image = "vmlinuz-{}".format(kernel_release)
+            self.initrd_image = "initramfs-{}.img".format(kernel_release)
+        self.cv_SYSTEM.goto_state(OpSystemState.OS)
+        res = self.c.run_command("cat /etc/os-release")
+        if "Ubuntu" in (res[0] or res[1]):
+            self.distro = "Ubuntu"
+        elif 'Red Hat' in (res[0] or res[1]):
+            self.distro = 'RHEL'
+        elif 'SLES' in (res[0] or res[1]):
+            self.distro = 'SLES'
+        else:
+            raise self.skipTest("Test currently supported only on Ubuntu, SLES and RHEL")
+ 
+    def get_raw_pty_console(self,cmd):
+        """
+        This function executes the command in raw_pty console 
+        """
+        try:
+            raw_pty = self.cv_SYSTEM.console.get_console()
+            raw_pty.sendline(cmd)
+            raw_pty.expect("login:", timeout=600)
+            log.info("{} passed(SUCCESS)".format(cmd))
+        except CommandFailed:
+            log.info("{} failed with exception.".format(cmd))
+
+    def kexec_load(self):
+        """
+        This function will test the kexec load
+        """
+        try:
+            self.c.run_command("kexec -l /boot/{} --initrd /boot/{} --append=\"`cat /proc/cmdline`\"".format(self.kernel_image,self.initrd_image))
+        except CommandFailed:
+            kexec_loaded_status = self.c.run_command("cat /sys/kernel/kexec_loaded")
+            if kexec_loaded_status[0] == "0":
+                log.info("Failed to load kernel image = \"%s\"", self.kernel_image)
+            raise OpTestError("kexec -l failed to load kernel")
+        
+    def kexec_unload(self):
+        """
+        This function will test the kexec unload
+        """
+        try:
+            self.c.run_command("kexec -u")
+        except CommandFailed:
+            kexec_loaded_status = self.c.run_command("cat /sys/kernel/kexec_loaded")
+            if kexec_loaded_status[0] == "1":
+                log.info("Failed to unload kernel image = \"%s\"", self.kernel_image)
+            raise OpTestError("Kexec -u  failed to unload")
```

Comment:
load and unload can be a single method ? as most of the code are same, except command

Reply:
kexec_load_unload can be called based on u/l input